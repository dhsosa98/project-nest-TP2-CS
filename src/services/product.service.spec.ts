import { ProductsService } from './products.service';

describe('ProductService', () => {
  let ProductService: ProductsService;

  beforeEach(() => {
    ProductService = new ProductsService();
  });

  const productSnapshot = {
    description: expect.any(String),
    id: expect.any(Number),
    image: expect.any(String),
    name: expect.any(String),
    price: expect.any(Number),
    stock: expect.any(Number),
  };

  it('return array Products', () => {
    ProductService.findAll().forEach((product) => {
      expect(product).toMatchSnapshot(productSnapshot);
    });
  });

  it('return element one', () => {
    expect(ProductService.findOne(1)).toMatchSnapshot(productSnapshot);
  });
});
