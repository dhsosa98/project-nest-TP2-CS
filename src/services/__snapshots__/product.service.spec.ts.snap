// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductService return array Products 1`] = `
Object {
  "description": Any<String>,
  "id": Any<Number>,
  "image": Any<String>,
  "name": Any<String>,
  "price": Any<Number>,
  "stock": Any<Number>,
}
`;

exports[`ProductService return array Products 2`] = `
Object {
  "description": Any<String>,
  "id": Any<Number>,
  "image": Any<String>,
  "name": Any<String>,
  "price": Any<Number>,
  "stock": Any<Number>,
}
`;

exports[`ProductService return element one 1`] = `
Object {
  "description": Any<String>,
  "id": Any<Number>,
  "image": Any<String>,
  "name": Any<String>,
  "price": Any<Number>,
  "stock": Any<Number>,
}
`;
